Update the auto-renew toggle to use a link on Purchase Details

dzver opened the pull request:
This component is used in several places. All other places should remain intact.

The change will replace the Toggle component with an On/Off button in the middle of the Auto-renew ON copy.

#### Proposed Changes

- The Auto-renew toggle should be visible as a link on the purchase details page.
- There should be no link if the user doesn't own the upgrade
- Domain Management and Email Management that also use this toggle should be unaffected

This is how it works with the purchase owner:
https://user-images.githubusercontent.com/82778/191526566-15ec4e91-48d5-4919-9acc-6a58fbd31202.mov

And this is how it looks with another admin account:
<img width="1066" alt="Screenshot 2022-09-21 at 17 13 47" src="https://user-images.githubusercontent.com/82778/191528163-54d65369-ebc2-40c6-bcad-ded37e80acd2.png">

#### Testing Instructions

* Buy a plan, domain, and email with the store sandbox
* Go to /me/purchases, then to a purchase detail of one of these upgrades (or any other)
* Try clicking on the new link and make sure the auto-renew goes on and off
* Go to Domain Management. Make sure the toggle is unchanged there
* Go to Email Management. Make sure the toggle is also unchanged
* Add another user to your test site. Make sure they see no clickable toggles anywhere

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

dzver committed:
Update the auto-renew toggle to use a link on Purchase Details

This component is used in several places. All other places should remain
intact.

The change will replace the Toggle component with an On/Off button in
the middle of the Auto-renew ON copy.

southp commented:
I've confirmed that it works as the test plan described. The code also looks good to me, but there is one accessibility issue: I can't focus on the new anchor-like button by tabbing, and the cursor doesn't change into a clickable pointer when hovering the mouse cursor onto it:

![on-anchor-pointer](https://user-images.githubusercontent.com/1842898/191647654-933e9b4d-7249-4f1e-8dfb-8556eb3e0271.gif)

I suppose the both might be the same issue, so the above screencast only demonstrates the cursor part :)

dzver committed:
There's no cursor and the tabbing doesn't work

Use a link button in order to make that work

southp reviewed (approved):
The fix works. LGTM 🚢

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7542517

Thank you @dzver for including a screenshot in the description! This is really helpful for our translators.

DavidRothstein reviewed (changes requested):
I think this one clearly needs more discussion: pebzTe-j5-p2#comment-471

dzver committed:
Rename the Modal heading

@DavidRothstein raised a concern about the flow not being immediately
clear with this heading. Rename it to make it more clear what it is

DavidRothstein reviewed (approved):
This approval does not actually mean I reviewed it (I didn't), but just trying to remove my "Changes requested" following the recent update to the PR -- this seems to be the only way to do it :smile:

DavidRothstein commented:
> recent update to the PR

Which changed the modal title that is shown after clicking "ON" from "Before you go..." to "Turn off auto-renew" (for future reference).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7542517).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7542517).
